lib/auth/webauthn/register.go

```diff
@@ -147,11 +147,15 @@ func (f *RegistrationFlow) Begin(ctx context.Context, user string, passwordless
 		if dev.GetU2F() != nil {
 			continue
 		}
-		// Skip resident/non-resident keys depending on whether it's a passwordless
-		// registration.
-		// Letting users have both allows them to "swap" between key types in the
-		// same device.
-		if webDev := dev.GetWebauthn(); webDev != nil && webDev.ResidentKey != passwordless {
+
+		// Let "upgrades" from non-resident to resident happen, but prevent
+		// authenticator "downgrades".
+		//
+		// Modern passkey implementations will "disobey" our MFA registrations and
+		// actually create passkeys, silently replacing the old passkey with the new
+		// "MFA" key, which can make Teleport confused (for example, by letting the
+		// "MFA" key be deleted because Teleport thinks the passkey still exists).
+		if webDev := dev.GetWebauthn(); webDev != nil && !webDev.ResidentKey && passwordless {
```

Comment:
nit/suggestion: Does resident/non-resident define whether it is passwordless? From the perspective of someone not familiar with this flow (me), the "downgrade" is not clear if it is `passwordless -> MFA` or `MFA -> passwordless`. Maybe mentioning key types of transition would help. For example: "Prevent device key type to transition from resident (e.g., passwordless) to non-resident (e.g., MFA)."

Reply:
Good call, I frequently get questions about this. Changed to:

```
// Let authenticator "upgrades" from non-resident (MFA) to resident
// (passwordless) happen, but prevent "downgrades" from resident to
// non-resident.
```